segmentation/include/vc/segmentation/OpticalFlowSegmentation.hpp

```diff
@@ -0,0 +1,177 @@
+
+    /** @brief Set whether to purge cache
+     */
+    void setPurgeCache(bool purge_c) { purge_cache_ = purge_c; }
+    /**@}*/
+
+    /** @brief Set how many slices should be cached
+     */
+    void setCacheSlices(int cache_slices) { nr_cache_slices_ = cache_slices; }
+
+    /** @brief Set how many slices should be cached
+     */
+    void setLineSmoothenByBrightness(int brightness) { smoothen_by_brightness_ = brightness; }
+
+    /**@{*/
+    /** @brief Compute the segmentation 1 Line */
+    std::vector<Voxel> computeCurve(FittedCurve currentCurve, Chain& currentVs, int zIndex);
+    /**@}*/
+
+    /**@{*/
+    /** @brief Compute the segmentation */
+    PointSet compute() override;
+    /**@}*/
+
+    /**@{*/
+    /** Debug: Shows intensity maps in GUI window */
+    void setVisualize(bool b) { visualize_ = b; }
+
+    /** Debug: Dumps reslices and intensity maps to disk */
+    void setDumpVis(bool b) { dumpVis_ = b; }
+
+    /** @brief Returns the maximum progress value */
+    [[nodiscard]] auto progressIterations() const -> size_t override;
+
+private:
+    /**
+     * @brief Estimate the normal to the curve at point index
+     * @param currentCurve Input curve
+     * @param index Index of point on curve
+     */
+    auto estimate_normal_at_index_(const FittedCurve& currentCurve, int index)
+        -> cv::Vec3d;
+
+    /**
+     * @brief Estimate the 2D normal to the curve at point index in the z plane
+     * @param curve Input curve
+     * @param index Index of point on curve
+     */
+    auto estimate_2d_normal_at_index_(const volcart::segmentation::FittedCurve& curve, int index)
+        -> cv::Vec2f;
+
+    /**
+     * @brief Calculate the mean pixel value in a window around a point
+     * @param integral_img Intergral image of slice
+     * @param pt Point around which to calculate mean
+     * @param window_size Size of window
+     */
+    auto get_mean_pixel_value(const cv::Mat& integral_img, const cv::Point& pt, int window_size)
+        -> float;
+
+    /**
+     * @brief Debug: Draw curve on slice image
+     * @param curve Input curve
+     * @param sliceIndex %Slice on which to draw
+     * @param particleIndex Highlight point at particleIndex
+     * @param showSpline Draw interpolated curve. Default only draws points
+     */
+    [[nodiscard]] auto draw_particle_on_slice_(
+        const FittedCurve& curve,
+        int sliceIndex,
+        int particleIndex = -1,
+        bool showSpline = false) const -> cv::Mat;
+
+    /** @brief Convert the internal storage array into a final PointSet */
+    auto create_final_pointset_(const std::vector<std::vector<Voxel>>& points)
+        -> PointSet;
+
+    /** Default minimum energy gradient */
+    constexpr static double DEFAULT_MIN_ENERGY_GRADIENT = 1e-7;
+
+    /** Target z-index */
+    int endIndex_{0};
+    /** Darker pixels are considered outside the sheet */
+    int outside_threshold_{80};
```

Comment:
For all of the new variables, check our [C++ naming conventions](https://gitlab.com/educelab/style-guides/-/blob/master/C++%20Style%20Guide.md#naming-conventions).

It's quick and easy for me to refactor these, so let me know if you don't have a good way to replace them.

Reply:
Tried to conform to the naming convention, might be good to have someone look over it again.